packages/frontend-plugin-api/api-report.md

```diff
@@ -913,6 +1050,42 @@ export interface ExtensionInput<
   extensionData: TExtensionData;
 }
 
+// @public
+export class ExtensionKind<
+  TOptions,
+  TInputs extends AnyExtensionInputMap,
+  TOutput extends AnyExtensionDataMap,
+  TConfig,
+> {
+  // (undocumented)
+  static create<
+    TOptions,
+    TInputs extends AnyExtensionInputMap,
+    TOutput extends AnyExtensionDataMap,
+    TConfig,
+  >(
+    options: CreateExtensionKindOptions<TOptions, TInputs, TOutput, TConfig>,
+  ): ExtensionKind<TOptions, TInputs, TOutput, TConfig>;
+  // (undocumented)
+  new(
+    instanceProperties: CreateExtensionKindInstanceOptions<
+      TOptions,
+      TInputs,
+      TOutput,
+      TConfig
+    >,
+  ): ExtensionDefinition<TConfig> & {
+    override: (
```

Comment:
I wonder if we should just leave this as a standalone `createExtensionOverride` or something like that instead? You'll need the extension definition either way, but I have a feeling it might be a simpler API to work with. I'll prolly play around with it a bit to see if there's any obvious benefits of either choice. In the end it's perhaps not a super important decision at this point and we might just ship either approach to try it out.

Another slightly more funky approach could also be to allow `createExtension` to receive an existing extension, in which case you build on top of that instead, i.e. override. Might be awkward but I could also see that being a pretty intuitive design.

Reply:
Let's tackle this in overrides.